DQMOffline/Trigger/plugins/HTMonitor.cc

```diff
@@ -0,0 +1,301 @@
+#include "DQMOffline/Trigger/plugins/HTMonitor.h"
+
+#include "CommonTools/TriggerUtils/interface/GenericTriggerEventFlag.h"
+
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+// -----------------------------
+//  constructors and destructor
+// -----------------------------
+
+HTMonitor::HTMonitor( const edm::ParameterSet& iConfig ) : 
+  folderName_             ( iConfig.getParameter<std::string>("FolderName") )
+  , metToken_             ( consumes<reco::PFMETCollection>      (iConfig.getParameter<edm::InputTag>("met")       ) )   
+  , jetToken_             ( mayConsume<reco::PFJetCollection>      (iConfig.getParameter<edm::InputTag>("jets")      ) )   
+  , eleToken_             ( mayConsume<reco::GsfElectronCollection>(iConfig.getParameter<edm::InputTag>("electrons") ) )   
+  , muoToken_             ( mayConsume<reco::MuonCollection>       (iConfig.getParameter<edm::InputTag>("muons")     ) )   
+  , vtxToken_             ( mayConsume<reco::VertexCollection>      (iConfig.getParameter<edm::InputTag>("vertices")      ) )
+  , ht_variable_binning_ ( iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >("htBinning") )
+  , ht_binning_          ( getHistoPSet   (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("htPSet")    ) )
+  , ls_binning_           ( getHistoLSPSet (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("lsPSet")     ) )
+  , num_genTriggerEventFlag_(new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("numGenericTriggerEventPSet"),consumesCollector(), *this))
+  , den_genTriggerEventFlag_(new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("denGenericTriggerEventPSet"),consumesCollector(), *this))
+  , metSelection_ ( iConfig.getParameter<std::string>("metSelection") )
+  , jetSelection_ ( iConfig.getParameter<std::string>("jetSelection") )
+  , eleSelection_ ( iConfig.getParameter<std::string>("eleSelection") )
+  , muoSelection_ ( iConfig.getParameter<std::string>("muoSelection") )
+  , jetSelection_HT_ ( iConfig.getParameter<std::string>("jetSelection_HT") )
+  , njets_      ( iConfig.getParameter<unsigned>("njets" )      )
+  , nelectrons_ ( iConfig.getParameter<unsigned>("nelectrons" ) )
+  , nmuons_     ( iConfig.getParameter<unsigned>("nmuons" )     )
+{
+
+  htME_variableBinning_.numerator   = nullptr;
```

Comment:
you can avoid these by initializing the variables within the struct definition itself.